### ISSUE TITLE
Received a NAK to the SET_CONFIG

### ISSUE THREAD

**Issue by ChrisK9EQ** [g4klx/MMDVMHost#520]
Using Pi-star, I've been getting a "Received a NAK to the SET_CONFIG command from modem" error whilst trying to run Pi-star with different input and output frequencies. Here's what I have learned:
1. The problem occurs in the YSF mode. Don't know about other modes. I have Self=0 and RemoteGateway=1. (These settings don't impact the problem.)

2. There are several places to set frequencies, but the only one that works reliably is Expert->MMDVM Host. Setting modem frequencies anywhere else causes problems. Obviously a Pi-star problem.

3. The recovery process requires the following steps:
3.1 In MMDVMHost set the Tx and Rx frequencies the same.
3.2. Switch to a different YSF Room. (After this point the NAK goes away.)
3.3 In MMDVMHost reset the Tx and Rx frequencies. I.e., 444025000 and 449025000. (Still working at this point, no NAK)
3.4. Switch back to the desired YSF reflector.

4.  The problem does not appear again until I make a small change. For example, changing the name from "pi-star" to "pi-star3" will cause the problem to occur again.

5. Reboots (software or power) have no impact on the problem. I.e., they don't fix anything.

6. Backing up the configuration files and restoring them causes the problem to occur.

Modem is MMDVM. The problem occurs with several modem boards. Occurs on several different RPi's with different (fresh) SD cards. (HS_Hat:v1.4.17, Pi-star 3.4.17 with recent upgrade/update.)

Unfortunately I can't convince the system to produce error messages that are more instructive than the following so I don't know what MMDVM is NAK'ing.

M: 2019-09-18 20:15:10.281 Opening the MMDVM
I: 2019-09-18 20:15:12.292 MMDVM protocol version: 1, description: MMDVM_HS_Hat-v1.4.17 20190529 14.7456MHz ADF7021 FW by CA6JAU GitID #cc451c4
E: 2019-09-18 20:15:12.313 Received a NAK to the SET_CONFIG command from the modem

For those who care, the rest of the log is below.

73,
Chris, K9EQ

I: 2019-09-18 20:15:10.279 This software is for use on amateur radio networks only,
I: 2019-09-18 20:15:10.279 it is to be used for educational purposes only. Its use on
I: 2019-09-18 20:15:10.279 commercial networks is strictly prohibited.
I: 2019-09-18 20:15:10.279 Copyright(C) 2015-2018 by Jonathan Naylor, G4KLX and others
M: 2019-09-18 20:15:10.279 MMDVMHost-20181107_Pi-Star is starting
M: 2019-09-18 20:15:10.279 Built 09:24:10 Nov 12 2018 (GitID #9444eca)
I: 2019-09-18 20:15:10.279 General Parameters
I: 2019-09-18 20:15:10.279     Callsign: K9EQ
I: 2019-09-18 20:15:10.279     Id: 3127144
I: 2019-09-18 20:15:10.279     Duplex: yes
I: 2019-09-18 20:15:10.279     Timeout: 240s
I: 2019-09-18 20:15:10.279     D-Star: disabled
I: 2019-09-18 20:15:10.279     DMR: disabled
I: 2019-09-18 20:15:10.279     YSF: enabled
I: 2019-09-18 20:15:10.280     P25: disabled
I: 2019-09-18 20:15:10.280     NXDN: disabled
I: 2019-09-18 20:15:10.280     POCSAG: disabled
I: 2019-09-18 20:15:10.280 Modem Parameters
I: 2019-09-18 20:15:10.280     Port: /dev/ttyAMA0
I: 2019-09-18 20:15:10.280     Protocol: uart
I: 2019-09-18 20:15:10.280     RX Invert: no
I: 2019-09-18 20:15:10.280     TX Invert: yes
I: 2019-09-18 20:15:10.280     PTT Invert: no
I: 2019-09-18 20:15:10.280     TX Delay: 100ms
I: 2019-09-18 20:15:10.280     RX Offset: 0Hz
I: 2019-09-18 20:15:10.280     TX Offset: 0Hz
I: 2019-09-18 20:15:10.280     RX DC Offset: 0
I: 2019-09-18 20:15:10.280     TX DC Offset: 0
I: 2019-09-18 20:15:10.280     RF Level: 100.0%
I: 2019-09-18 20:15:10.281     DMR Delay: 0 (0.0ms)
I: 2019-09-18 20:15:10.281     RX Level: 50.0%
I: 2019-09-18 20:15:10.281     CW Id TX Level: 50.0%
I: 2019-09-18 20:15:10.281     D-Star TX Level: 50.0%
I: 2019-09-18 20:15:10.281     DMR TX Level: 50.0%
I: 2019-09-18 20:15:10.281     YSF TX Level: 50.0%
I: 2019-09-18 20:15:10.281     P25 TX Level: 50.0%
I: 2019-09-18 20:15:10.281     NXDN TX Level: 50.0%
I: 2019-09-18 20:15:10.281     POCSAG TX Level: 50.0%
I: 2019-09-18 20:15:10.281     RX Frequency: 444025000Hz (444025000Hz)
I: 2019-09-18 20:15:10.281     TX Frequency: 449025000Hz (449025000Hz)
M: 2019-09-18 20:15:10.281 Opening the MMDVM
I: 2019-09-18 20:15:12.292 MMDVM protocol version: 1, description: MMDVM_HS_Hat-v1.4.17 20190529 14.7456MHz ADF7021 FW by CA6JAU GitID #cc451c4
E: 2019-09-18 20:15:12.313 Received a NAK to the SET_CONFIG command from the modem

**Comment by ChrisK9EQ** [comment]
Solved the problem. It is a Pi-star issue. They were changing DUPLEX=1 whenever Tx != Rx even though the modem board was not capable of duplex operation. Hence the NAK. Removing code in /var/www/dashboard/admin/configure.php where it was setting DUPLEX or SIMPLEX based on frequency solved the problem.

This explains why I couldn't make any changes after I set the offset in frequencies. Another bug in the code causes the duplex change to take place when the services are restarted after the service restart where the initialization files were changed.

This issue can be closed with no action.
